src/graph.rs

```diff
@@ -2077,110 +2079,133 @@ pub(crate) fn parse_esm_module_from_module_info(
 
   // Analyze ES dependencies
   for desc in module_info.dependencies {
-    let is_import_or_export_type = matches!(
-      desc.kind,
-      DependencyKind::ImportType | DependencyKind::ExportType
-    );
-    if is_import_or_export_type && !graph_kind.include_types() {
-      continue; // skip
-    }
+    let (import, leading_comments) = match desc {
+      DependencyDescriptor::Static(desc) => {
+        let is_import_or_export_type = matches!(
+          desc.kind,
+          DependencyKind::ImportType | DependencyKind::ExportType
+        );
+        if is_import_or_export_type && !graph_kind.include_types() {
+          continue; // skip
+        }
+        let range = Range::from_position_range(
+          module.specifier.clone(),
+          desc.specifier_range.clone(),
+        );
+
+        (
+          Import {
+            specifier: desc.specifier,
+            kind: match is_import_or_export_type {
+              true => ImportKind::TsType,
+              false => ImportKind::Es,
+            },
+            range,
+            is_dynamic: false,
+            attributes: desc.import_attributes,
+          },
+          desc.leading_comments,
+        )
+      }
+      DependencyDescriptor::Dynamic(desc) => {
+        // todo(#275): resolve more dynamic imports here
+        let specifier = match &desc.argument {
+          DynamicArgument::String(text) => text.to_string(),
+          DynamicArgument::Template(_) | DynamicArgument::Expr => continue,
+        };
```

Comment:
In the future, we'll be able to implement https://github.com/denoland/deno_graph/issues/275 here.